Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -584,7 +584,8 @@ public void ClearStateCache(VBProject project, bool notifyStateChanged = false)
 
             if (notifyStateChanged)
             {
-                OnStateChanged();
+                OnStateChanged(ParserState.ResolvedDeclarations);   // trigger test explorer and code explorer updates
+                OnStateChanged(ParserState.Ready);   // trigger find all references &c. updates
```

Comment:
Isn't that kinda evil? Shouldn't we just notify TE and CE when parsing finished and actually passed over `ResolvedDeclarations` instead of manually firing the events twice?